src/basic/macro.h

```diff
@@ -380,11 +374,11 @@ assert_cc(sizeof(dummy_t) == 0);
                 _q && _q > (base) ? &_q[-1] : NULL;      \
         })
 
-/* Iterate through each variadic arg. All must be the same type as 'entry' or must be implicitly
+/* Iterate through each argument passed. All must be the same type as 'entry' or must be implicitly
  * convertible. The iteration variable 'entry' must already be defined. */
-#define VA_ARGS_FOREACH(entry, ...)                                     \
-        _VA_ARGS_FOREACH(entry, UNIQ_T(_entries_, UNIQ), UNIQ_T(_current_, UNIQ), UNIQ_T(_va_sentinel_, UNIQ), ##__VA_ARGS__)
-#define _VA_ARGS_FOREACH(entry, _entries_, _current_, _va_sentinel_, ...)         \
+#define FOREACH_ARGUMENT(entry, ...)                                     \
```

Comment:
Your commit message claims the existing naming is confusing since this macro has nothing to do with va_arg, but it clearly does use the macro expansion form of it here via "..." and `__VA_ARGS__` et al below.  It's a variadic macro, I fail to see the confusion, so this commit seems superfluous to me.

Reply:
The name might suggest that it is _used to handle_ variable argument lists, rather than that it is a varadic macro. And the naming should generally describe the usage instead of the implementation IMHO.